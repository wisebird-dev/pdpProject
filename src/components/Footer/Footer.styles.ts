import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(({ palette }) => ({
    footer: {
        position: 'absolute',
        textAlign: 'center',
        width: '100%',
        bottom: 0,
        right: 0,
        backgroundColor: palette.secondary.main,
    },
}));

export default useStyles;
