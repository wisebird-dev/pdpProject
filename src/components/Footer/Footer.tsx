import React from 'react';

import { Typography } from '@components';
import {
    DEV_PROFILE_URL,
    GIT_PROFILE_URL,
    GIT_REPO_URL,
    LINKEDIN_PROFILE_URL,
    STACK_OVERFLOW_PROFILE_URL,
} from '@core';

import useStyles from './Footer.styles';
import { LinkExternal, Copyright } from './components';

const Footer: React.FC = () => {
    const classes = useStyles();

    return (
        <article>
            <footer className={classes.footer}>
                <div className={classes.container}>
                    <div className={classes.containerItem}>
                        <Typography component="h4" variant="h6">
                            Social
                        </Typography>
                        <LinkExternal
                            url={LINKEDIN_PROFILE_URL}
                            text="linkedIn"
                        />
                        <LinkExternal
                            url={STACK_OVERFLOW_PROFILE_URL}
                            text="stack overflow"
                        />
                        <LinkExternal url={DEV_PROFILE_URL} text="DEV.to" />
                    </div>
                    <div className={classes.containerItem}>
                        <Typography component="h4" variant="h6">
                            FAQ
                        </Typography>
                        <Typography variant="body1">
                            about this project
                        </Typography>
                    </div>
                    <div className={classes.containerItem}>
                        <Typography component="h4" variant="h6">
                            github
                        </Typography>
                        <LinkExternal
                            url={GIT_REPO_URL}
                            text="project repository"
                        />
                        <LinkExternal
                            url={GIT_PROFILE_URL}
                            text="github profile"
                        />
                    </div>
                </div>
                <Copyright />
            </footer>
        </article>
    );
};

export default Footer;
