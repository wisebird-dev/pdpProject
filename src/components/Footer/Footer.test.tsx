import React from 'react';

import { renderWithProviders } from '@testUtils';

import Footer from './Footer';

describe('Footer component', () => {
    it('matches snapshot', () => {
        const { container } = renderWithProviders(<Footer />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
