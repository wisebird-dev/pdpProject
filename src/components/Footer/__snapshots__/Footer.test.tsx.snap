// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot 1`] = `
<article>
  <footer
    class="makeStyles-footer-1"
  >
    <div
      class="makeStyles-container-2"
    >
      <div
        class="makeStyles-containerItem-3"
      >
        <h4
          class="MuiTypography-root makeStyles-section-4 MuiTypography-body1"
        >
          Social
        </h4>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
          href="https://www.linkedin.com/in/spyros-goulas"
          rel="noopener"
          target="_blank"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            linkedIn
          </p>
        </a>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
          href="https://stackoverflow.com/users/6950690/%ce%a3%cf%80%cf%8d%cf%81%ce%bf%cf%82-%ce%93%ce%bf%cf%8d%ce%bb%ce%b1%cf%82"
          rel="noopener"
          target="_blank"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            stack overflow
          </p>
        </a>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
          href="https://dev.to/sgoulas"
          rel="noopener"
          target="_blank"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            DEV.to
          </p>
        </a>
      </div>
      <div
        class="makeStyles-containerItem-3"
      >
        <h4
          class="MuiTypography-root makeStyles-section-4 MuiTypography-body1"
        >
          FAQ
        </h4>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          about this project
        </p>
      </div>
      <div
        class="makeStyles-containerItem-3"
      >
        <h4
          class="MuiTypography-root makeStyles-section-4 MuiTypography-body1"
        >
          github
        </h4>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
          href="https://github.com/sgoulas/pdpProject"
          rel="noopener"
          target="_blank"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            project repository
          </p>
        </a>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
          href="https://github.com/sgoulas"
          rel="noopener"
          target="_blank"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            github profile
          </p>
        </a>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-middle"
    />
    <p
      class="MuiTypography-root makeStyles-text-5 MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-alignRight"
    >
      made by Spyros Goulas
    </p>
  </footer>
</article>
`;
